In [43]:
#import relevant libraries
import pandas as pd

In [44]:
#reading out data files
data=pd.read_csv('FoodBalanceSheets_E_Africa_NOFLAG.csv')

In [45]:
#checking columns
data.columns

Index(['Area Code', 'Area', 'Item Code', 'Item', 'Element Code', 'Element',
       'Unit', 'Y2014', 'Y2015', 'Y2016', 'Y2017', 'Y2018'],
      dtype='object')

In [4]:
data.describe()

Area Code     Item Code  Element Code          Y2014          Y2015  \
count  60943.000000  60943.000000  60943.000000   59354.000000   59395.000000   
mean     134.265576   2687.176706   3814.856456     134.196282     135.235966   
std       72.605709    146.055739   2212.007033    1567.663696    1603.403984   
min        4.000000   2501.000000    511.000000   -1796.000000   -3161.000000   
25%       74.000000   2562.000000    684.000000       0.000000       0.000000   
50%      136.000000   2630.000000   5142.000000       0.090000       0.080000   
75%      195.000000   2775.000000   5511.000000       8.340000       8.460000   
max      276.000000   2961.000000   5911.000000  176405.000000  181137.000000   

               Y2016          Y2017          Y2018  
count   59408.000000   59437.000000   59507.000000  
mean      136.555222     140.917765     143.758381  
std      1640.007194    1671.862359    1710.782658  
min     -3225.000000   -1582.000000   -3396.000000  
25%         0.000000       0.000000       0.000000  
50%         0.080000       0.100000       0.070000  
75%         8.430000       9.000000       9.000000  
max    185960.000000  190873.000000  195875.000000

In [46]:
data.head(7)

Area Code     Area  Item Code              Item  Element Code  \
0          4  Algeria       2501        Population           511   
1          4  Algeria       2501        Population          5301   
2          4  Algeria       2901       Grand Total           664   
3          4  Algeria       2901       Grand Total           674   
4          4  Algeria       2901       Grand Total           684   
5          4  Algeria       2903  Vegetal Products           664   
6          4  Algeria       2903  Vegetal Products           674   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   
5           Food supply (kcal/capita/day)  kcal/capita/day   2932.00   
6  Protein supply quantity (g/capita/day)     g/capita/day     67.14   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28  
5   2958.00   2941.00   2921.00   2932.00  
6     67.38     67.37     66.11     67.10

#### What is the total sum of Animal Fat produced in 2014 and 2017 respectively?
Hint: Perform a groupby sum aggregration on 'item'

In [19]:
#check if the animal fat is present in item
data['Item'].unique()

array(['Population', 'Grand Total', 'Vegetal Products', 'Animal Products',
       'Cereals - Excluding Beer', 'Wheat and products',
       'Rice and products', 'Barley and products', 'Maize and products',
       'Rye and products', 'Oats', 'Millet and products',
       'Sorghum and products', 'Cereals, Other', 'Starchy Roots',
       'Cassava and products', 'Potatoes and products', 'Sweet potatoes',
       'Yams', 'Roots, Other', 'Sugar Crops', 'Sugar cane', 'Sugar beet',
       'Sugar & Sweeteners', 'Sugar (Raw Equivalent)',
       'Sweeteners, Other', 'Honey', 'Pulses', 'Beans', 'Peas',
       'Pulses, Other and products', 'Treenuts', 'Nuts and products',
       'Oilcrops', 'Soyabeans', 'Groundnuts (Shelled Eq)',
       'Sunflower seed', 'Rape and Mustardseed', 'Cottonseed',
       'Coconuts - Incl Copra', 'Sesame seed',
       'Olives (including preserved)', 'Oilcrops, Other',
       'Vegetable Oils', 'Soyabean Oil', 'Groundnut Oil',
       'Sunflowerseed Oil', 'Rape and Mustard Oil

The Animal fat item is present

In [22]:
#create a dataframe that groupby Item with sum aggregration
data1 = data.groupby(by='Item').sum()

In [23]:
data1.head()

Area Code  Item Code  Element Code      Y2014      Y2015  \
Item                                                                            
Alcohol, Non-Food        38447     752497       1515090    2403.00    2180.00   
Alcoholic Beverages      75016    1649136       2171381  102410.11   98783.72   
Animal Products          18060     397035         90990   11935.65   11811.73   
Animal fats              85279    1882494       2567303  209460.54  200675.72   
Apples and products      69280    1352989       1920418    9499.23   10559.15   

                         Y2016      Y2017      Y2018  
Item                                                  
Alcohol, Non-Food      2223.00    2348.00    2293.00  
Alcoholic Beverages   96958.75   95581.06   97847.27  
Animal Products       11661.69   11547.65   11578.61  
Animal fats          183314.22  269617.53  269648.27  
Apples and products    9853.26   10198.90    9640.51

From the dataframe data1 the sum of Animal fats for Y2014 is 209460.54 while that of Y2017 is 269617.53

#### What is the mean and the standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [35]:
#finding the mean for the whole dataset for 2015
data['Y2015'].mean()

135.23596649549236

In [28]:
#finding the standard deviation for the 2015
data['Y2015'].std()

1603.4039839423301

The mean and the standard deviation is 135.23596649549236 and 1603.4039839423301

#### What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [36]:
#total number of miissing data in 2016
data['Y2016'].isnull().sum()

1535

In [42]:
#To calculate the percentage missing 
percent_miss_Y2016 = data['Y2016'].isnull().sum()/data['Y2016'].value_counts().sum()
percent_miss_Y2016 = percent_miss_Y2016*100
percent_miss_Y2016

2.583827093994075

Total number of missing data is 1535 and percentage is 2.58

#### What is the highest correlation with 'Element Code'?

In [55]:
#to view the corr of all features with 'Element Code'
data.corr()['Element Code'][3:]

Y2014    0.024457
Y2015    0.023889
Y2016    0.023444
Y2017    0.024254
Y2018    0.024279
Name: Element Code, dtype: float64

Y2014 shows the highest correlation with Element Code

#### What year has the highest sum of Import Quantity

In [56]:
# lemme check out the location of Import Quantity
data['Element'].unique()

array(['Total Population - Both sexes', 'Domestic supply quantity',
       'Food supply (kcal/capita/day)',
       'Protein supply quantity (g/capita/day)',
       'Fat supply quantity (g/capita/day)', 'Production',
       'Import Quantity', 'Stock Variation', 'Export Quantity', 'Feed',
       'Seed', 'Losses', 'Processing', 'Other uses (non-food)',
       'Residuals', 'Food', 'Food supply quantity (kg/capita/yr)',
       'Tourist consumption'], dtype=object)

Import Quanity is present

Hint: Perform a groupby operation on 'Element' and use the resulting dataframe to answer the question 

In [60]:
#crating a dataframe of groupedby element
data2 = data.groupby(by = 'Element').sum()

In [68]:
data2.iloc[7:8,3:]

Y2014      Y2015      Y2016      Y2017      Y2018
Element                                                               
Import Quantity  274144.48  267018.46  286582.78  294559.09  287997.09

Y2017 has the highest sum of Import Quantity

#### What is the total number of the sum of Production in 2014

Hint: Perform a groupby operation on 'Element' and use the resulting dataframe to answer the question

In [69]:
#check whether the Production is included in Element
data['Element'].unique()

array(['Total Population - Both sexes', 'Domestic supply quantity',
       'Food supply (kcal/capita/day)',
       'Protein supply quantity (g/capita/day)',
       'Fat supply quantity (g/capita/day)', 'Production',
       'Import Quantity', 'Stock Variation', 'Export Quantity', 'Feed',
       'Seed', 'Losses', 'Processing', 'Other uses (non-food)',
       'Residuals', 'Food', 'Food supply quantity (kg/capita/yr)',
       'Tourist consumption'], dtype=object)

Production is present

In [70]:
#creating a dataframe of grouped by element
data3 = data.groupby(by = 'Element').sum()

In [73]:
data3.iloc[11:12,:]

Area Code  Item Code  Element Code       Y2014       Y2015  \
Element                                                                  
Production     526751   10450053      21388191  1931287.75  1947019.39   

                 Y2016       Y2017       Y2018  
Element                                         
Production  1943537.15  2030056.89  2075072.89

Total Production for Y2014 is  1931287.75

#### Which of these elements had the highest sum in 2018?

Hint: Select columns 'Y2018' and 'Element', Perform a groupby operation on 'Element' on the selected dataframe and answer the question.

In [79]:
#Selcting the two columns (Y2018 and Element)
data4 = data[['Element','Y2018']]
#perform a groupby operation on Element
data5 = data4.groupby(by = 'Element').sum()

In [97]:
data5.sort_values(by = 'Y2018',ascending = False).iloc[:1,:]

Y2018
Element                            
Domestic supply quantity  2161192.1

Domestic supply quantity has the highest sum in 2018

#### Which of these elements has the 3rd lowest sum in 2018?

Hint: Select columns 'Y2018' and 'Element', Perform a groupby operation on 'Element' on the selected dataframe and answer the question.

In [98]:
#since i already have a dataframe done for that 
#recalling the data frame 
data5.head(2)

Y2018
Element                            
Domestic supply quantity  2161192.1
Export Quantity            181594.8

In [104]:
data5.sort_values(by = 'Y2018',ascending = True)

Y2018
Element                                           
Tourist consumption                          90.00
Fat supply quantity (g/capita/day)        10258.69
Protein supply quantity (g/capita/day)    11833.56
Stock Variation                           20577.91
Seed                                      25263.14
Residuals                                 34864.00
Food supply quantity (kg/capita/yr)       49056.85
Other uses (non-food)                     91300.97
Losses                                   163902.00
Export Quantity                          181594.80
Feed                                     233489.68
Import Quantity                          287997.09
Processing                               308429.00
Food supply (kcal/capita/day)            455261.00
Total Population - Both sexes           1140605.00
Food                                    1303841.28
Production                              2075072.89
Domestic supply quantity                2161192.10

Protein supply quantity (g/capita/day) has the 3rd lowest sum in Y2018

#### What is the total Import Quantity in Algeria in 2018 

In [145]:
data[data['Element']=='Import Quantity'].loc[data['Area']=='Algeria'].sum()

Area Code                                                     448
Area            AlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlgeriaAlge...
Item Code                                                  301030
Item            Cereals - Excluding BeerWheat and productsRice...
Element Code                                               628432
Element         Import QuantityImport QuantityImport QuantityI...
Unit            1000 tonnes1000 tonnes1000 tonnes1000 tonnes10...
Y2014                                                    34940.12
Y2015                                                    37841.91
Y2016                                                    36736.08
Y2017                                                    36059.29
Y2018                                                    36238.29
dtype: object

The total Import Quantity in Algeria in 2018 is 36238.29

#### What is the total number of the unique countries in the dataset

In [129]:
data['Area'].unique()

array(['Algeria', 'Angola', 'Benin', 'Botswana', 'Burkina Faso',
       'Cabo Verde', 'Cameroon', 'Central African Republic', 'Chad',
       'Comoros', 'Congo', "C�te d'Ivoire", 'Djibouti', 'Egypt',
       'Eswatini', 'Ethiopia', 'Ethiopia PDR', 'Gabon', 'Gambia', 'Ghana',
       'Guinea', 'Guinea-Bissau', 'Kenya', 'Lesotho', 'Liberia',
       'Madagascar', 'Malawi', 'Mali', 'Mauritania', 'Mauritius',
       'Morocco', 'Mozambique', 'Namibia', 'Niger', 'Nigeria', 'Rwanda',
       'Sao Tome and Principe', 'Senegal', 'Seychelles', 'Sierra Leone',
       'South Africa', 'Sudan', 'Sudan (former)', 'Togo', 'Tunisia',
       'Uganda', 'United Republic of Tanzania', 'Zambia', 'Zimbabwe'],
      dtype=object)

In [130]:
data['Area'].nunique()

49

There 49 unique countries in the dataset